### PR TITLE
docs: fixes for agent config and agent labels [DET-3958][DET-2372]

### PR DESCRIPTION
**Pull request by neilconway**
* Document `resources.agent_label` in experiment/command config
* Document TLS options in agent config
* Add support for proxy-related settings to agent CLI
* Clarify docs for setting master and agent settings via env variables
* Add FAQ on using agent labels to influence task placement

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

### CODE REVIEW

File: docs/faq.txt [hunk 1]
```diff
@@ -164,3 +164,20 @@ Yes! For an example of this check out the :download:`mnist-GAN
 automatically configures the location for writing TensorBoards. Users can also
 directly use ``torch.utils.tensorboard.SummaryWriter`` as shown in the snippet
 above.
+
+
+Scheduling
+----------
+
+How can I control which agents a task is scheduled on?
```
Comment (davidhershey) [inline docs/faq.txt:172]: Maybe phrase this more in the Determined novice way -- something like
"My cluster has machines with two different types of GPUs, how can I run a task on a specific machine?"

File: docs/reference/cluster-config.txt [hunk 2]
```diff
@@ -543,16 +570,34 @@ The master supports the following configuration settings:
...
   command.
 - ``http_proxy``: The HTTP proxy address for the agent's containers.
 - ``https_proxy``: The HTTPS proxy address for the agent's containers.
 - ``ftp_proxy``: The FTP proxy address for the agent's containers.
-- ``no_proxy``: The addresses that the container should not proxy.
+- ``no_proxy``: The addresses that the agent's containers should not proxy.
+- ``security``: Security-related configuration settings.
+
+  - ``tls``: TLS-related configuration settings.
+
```
Comment (dzhu) [inline docs/reference/cluster-config.txt:598]: non-blocking: Delete whitespace.

File: docs/reference/cluster-config.txt [hunk 3]
```diff
@@ -23,29 +23,53 @@ agent using a custom configuration file:
...
+    -e DET_MASTER_PORT=8080 \
+    determinedai/determined-agent
+
+The equivalent behavior can be achieved using command-line options:
+
+.. code::
+
+  determined-agent run --master-host=127.0.0.1 --master-port=8080
+
+The same behavior applies to master configuration settings as well. For example,
```
[the remark this answers is not in the document]
Comment (neilconway) [reply docs/reference/cluster-config.txt:62]: I agree we should make this more consistent -- to both enable all options to be set via command-line args and env variables, and to reduce any needless inconsistency between the agent and the master.

In the meantime, do you think this statement is true enough or do we need to add caveats to avoid confusion?

File: docs/faq.txt [hunk 4]
```diff
@@ -118,9 +118,25 @@ Please check if agent machines can access each other outside of Determined
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This can be done by assigning a label to certain agents in the cluster, and then
+using the :ref:`resources.agent_label <exp-config-agent_label>` setting in
+experiment, notebook, or command configurations. Each agent can have at most one
+label, which can be set via the :ref:`agent configuration
+<agent-configuration>`. Similarly, each task can have at most one
+``agent_label``. If a task is configured with an ``agent_label``, it will *only*
+be launched on agents that have that label. Similarly, if an agent has a label,
+it will never be assigned tasks that do not have an ``agent_label`` or that
```
[the remark this answers is not in the document]
Comment (neilconway) [reply docs/faq.txt:137]: This is accurate, but I felt like being a little more verbose to clarify the empty-label case was worth it.

File: docs/reference/cluster-config.txt [hunk 5]
```diff
@@ -533,26 +562,46 @@ The master supports the following configuration settings:
   - ``port``: The database port to use. (*Required*)
   - ``name``: The database name to use. (*Required*)
 
-- ``telemetry``: Specifies whether we collect anonymous information about the
-  usage of Determined.
+- ``telemetry``: Specifies whether we collect and report anonymous information
```
Comment (dzhu) [inline docs/reference/cluster-config.txt:565]: non-blocking: This ought to be more like "Configures how we collect...".

**Comment by dzhu** [comment]
Oops, I was just in the middle of writing those when you merged. This is fine.
